test/metabase/api/field_test.clj

```diff
@@ -742,70 +742,81 @@
     (when-not (mysql-test/is-mariadb? (u/id (mt/db)))
       (mt/dataset json
-        ;; Create a new database with the same details as the json dataset, with json unfolding enabled by default
+        ;; Create a new database with the same details as the json dataset, with json unfolding enabled
         (let [database (t2/select-one Database :id (mt/id))]
           (mt/with-temp* [Database [database {:engine driver/*driver*, :details (assoc (:details database) :json-unfolding true)}]]
             (mt/with-db database
               ;; Sync the new database
               (sync/sync-database! database)
               (let [field (t2/select-one Field :id (mt/id :json :json_bit))
-                    set-json-unfolding! (fn [v]
-                                          (mt/user-http-request :crowberto :put 200 (format "field/%d" (mt/id :json :json_bit))
-                                                                (assoc field :json_unfolding v)))
+                    get-field (fn [] (t2/select-one Field :id (mt/id :json :json_bit)))
+                    get-database (fn [] (t2/select-one Database :id (mt/id)))
+                    set-json-unfolding-for-field! (fn [v]
+                                                    (mt/user-http-request :crowberto :put 200 (format "field/%d" (mt/id :json :json_bit))
+                                                                          (assoc field :json_unfolding v)))
+                    set-json-unfolding-for-db! (fn [v]
+                                                 (let [updated-db (into {} (assoc-in database [:details :json-unfolding] v))]
+                                                   (mt/user-http-request :crowberto :put 200 (format "database/%d" (:id database))
+                                                                         updated-db)))
                     nested-fields          (fn []
                                              (->> (t2/select Field :table_id (mt/id :json) :active true :nfc_path [:not= nil])
                                                   (filter (fn [field] (= (first (:nfc_path field)) "json_bit")))))]
-                (testing "json-unfolding is enabled by default"
+                (testing "json_unfolding is enabled by default at the field level"
                   (is (true? (:json_unfolding field))))
                 (testing "nested fields are present since json unfolding is enabled by default"
                   (is (seq (nested-fields))))
-                (testing "nested fields are removed when json unfolding is disabled"
-                  (set-json-unfolding! false)
-                  (sync/sync-database! database)
+                (testing "nested fields are removed when json unfolding is disabled for the DB"
+                  (set-json-unfolding-for-db! false)
+                  (sync/sync-database! (get-database))
                   (is (empty? (nested-fields))))
```

Comment:
should we have a check that all json fields are folded? this test currently only checks the `json-bit` field.

Reply:
I wouldn't consider that a valuable addition. In what world could changing the database-level setting only affect a subset of JSON fields in a database?